jujugui/static/gui/src/app/components/deployment/add-credentials/add-credentials.js

```diff
@@ -32,22 +32,53 @@ YUI.add('deployment-add-credentials', function() {
     },
 
     /**
+      Validate the form fields.
+
+      @method _validate
+      @returns {Boolean} Whether the form is valid.
+    */
+    _validate: function() {
```

Comment:
This validation logic seems fairly copy-n-paste between the various components; could it be pulled into an extension?